### PR TITLE
feat: Rename all SHA3 opcodes to KECCAK256

### PR DESCRIPTION
**Pull request by tungbq**
Closes: #513

### CODE REVIEW

**Comment by rakita** [comment]
@tungbq thank you for taking this!

There is one function `sha3` that could use renaming.

**Comment by tungbq** [comment]
> @tungbq thank you for taking this!
> 
> There is one function `sha3` that could use renaming.

Thanks @rakita for reviewing. I've updated the new commits to update sha3 function.
Please check if it meets the requirements. Thanks

File: crates/precompile/Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ once_cell = "1.17"
 ripemd = { version = "0.1", default-features = false }
 secp256k1 = { version = "0.27.0", default-features = false, features = ["alloc", "recovery"], optional = true }
 sha2 = { version = "0.10.5", default-features = false }
-sha3 = { version = "0.10.7", default-features = false }
```
[the remark this answers is not in the document]
Comment (tungbq) [reply crates/precompile/Cargo.toml]: Updated

**Review by rakita — COMMENTED** [review]
there are failing test

**Comment by tungbq** [comment]
> there are failing test

Updated

**Comment by rakita** [comment]
Looks nice, will merge it after CI finishes

**Pull request merged**